src/components/hack-details.js

```diff
@@ -30,37 +34,72 @@ const useStyles = makeStyles(theme => ({
   },
   chipMArgin: {
     marginRight: theme.spacing(1)
+  },
+  marginTop: {
+    marginTop: theme.spacing(2)
+  },
+  marginFix: {
+    marginLeft: theme.spacing(0.1),
+    marginRight: theme.spacing(0.1),
+    paddingTop: theme.spacing(1),
+    paddingBottom: theme.spacing(1)
   }
 }));
 
 export function HackDetails(props) {
   const classes = useStyles();
 
-  const [data, setData] = useState(null);
+  const [displayData, setDsiplayData] = useState(null);
+  const [updateData, setUpdateData] = useState({});
+  const [edit, setEdit] = useState(false);
 
   useEffect(() => {
-    //console.log(props.match.params);
-
     if (props.match !== undefined) {
       Axios.get(
-        UrlJoin(process.env.REACT_APP_API_URL,"hackdetail/",props.match.params.id)
+        UrlJoin(
+          process.env.REACT_APP_API_URL,
+          "hackdetail/",
+          props.match.params.id
+        )
       ).then(res => {
-        setData(res.data)
+        setDsiplayData(res.data);
       });
     }
   }, [props.match]);
 
-  const joinHack = event => {
+  const joinHack = () => {
     const object = {
       hackId: props.match.params.id,
       userId: props.userId
     };
-    Axios.post(UrlJoin(process.env.REACT_APP_API_URL,"joinhack"), object).then(res => {
-      setData(res.data);
+    Axios.post(UrlJoin(process.env.REACT_APP_API_URL, "joinhack"), object).then(
+      res => {
+        setDsiplayData(res.data);
+      }
+    );
+  };
+
+  const handleOnChange = e => {
+    const { name, value } = e.target;
+    setUpdateData({ ...updateData, [name]: value });
+  };
+
+  const onSubmitUpdate = e => {
+    e.preventDefault();
+    Axios.post(UrlJoin(process.env.REACT_APP_API_URL, "edithack"), {
+      ...updateData,
+      hackId: displayData._id
+    }).then(res => {
+      setEdit(false);
```

Comment:
should this be set before we do the post? or have a submitting state that disabled the edits?